bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -1224,22 +1224,37 @@ private CompletableFuture<?> publishSuspectedLedgersAsync(Collection<String> mis
         LOG.info("Following ledgers: {} of bookie: {} are identified as underreplicated", ledgers, missingBookies);
         numUnderReplicatedLedger.registerSuccessfulValue(ledgers.size());
         LongAdder underReplicatedSize = new LongAdder();
+
         FutureUtils.processList(
                 Lists.newArrayList(ledgers),
-                ledgerId ->
-                    ledgerManager.readLedgerMetadata(ledgerId).whenComplete((metadata, exception) -> {
+                ledgerId -> {
+                    try {
+                        openLedgerNoRecoverySemaphore.acquire();
+                    } catch (InterruptedException e) {
+                        LOG.error("Unable to acquire open ledger operation semaphore ", e);
+                        Thread.currentThread().interrupt();
```

Comment:
It won't block the processing stream.

Reply:
Why? Since `processList` is processing  the list of items one by one